shoop/core/pricing/__init__.py

```diff
@@ -116,26 +120,9 @@ def get_context_from_request(self, request):
     def get_context_from_data(self, **context_data):
         return self.pricing_context_class(**context_data)
 
-    def get_price(self, context, product_id, quantity=1):
-        """
-        Get context-specific Price for the given product and quantity.
-
-        May return TaxlessPrice or TaxfulPrice.
-
-        :type context: PricingContext
-        :rtype: Price
-        """
-        return TaxlessPrice(0)
-
-    def get_base_price(self, product_id):
-        """
-        Get base Price for the given product.
-
-        May return TaxlessPrice or TaxfulPrice.
-
-        :rtype: Price
-        """
-        return TaxlessPrice(0)
+    @abc.abstractmethod
+    def get_price_info(self, context, product, quantity=1):
```

Comment:
So yeah, as mentioned, this method should accept a product ID as product as well.